redash/query_runner/google_spreadsheets.py

```diff
@@ -100,18 +100,28 @@ def __init__(self, worksheet_title):
 def parse_query(query):
     values = query.split("|")
     key = values[0]  # key of the spreadsheet
-    worksheet_num_or_title = 0  # A default value for when a number of inputs is invalid
+    worksheet_num = 0  # A default value for when a number of inputs is invalid
+    worksheet_title = '' # No title initially
     if len(values) == 2:
         s = values[1].strip()
         if len(s) > 0:
             if re.match(r"^\"(.*?)\"$", s):
                 # A string quoted by " means a title of worksheet
-                worksheet_num_or_title = s[1:-1]
-            else:
+                worksheet_title = s[1:-1]
+            elif re.match(r"^\d+$", s):
                 # if spreadsheet contains more than one worksheet - this is the number of it
-                worksheet_num_or_title = int(s)
-
-    return key, worksheet_num_or_title
+                worksheet_num = int(s)
+            else:
+                # To match for non-quoted one-word file name
+                if len(s.split()) == 1 :
+                    worksheet_title = s
+            if worksheet_title != '':
+                return key, worksheet_title
+            else:
+                return key, worksheet_num
```

Comment:
The function returning a string or number for the second element of the return tuple is a code smell - we should see if we can separate this out into two functions that handle each case separately, or have both the string and integer handling happen via integers only (or strings only).

Reply:
Also, by simplifying this we can also fix `parse_worksheet` to be simpler too.